Automattic/simplenote-android
# Add RTL Support

### Fix
Add support for right-to-left languages and layouts.  Fortunately, nearly all of the layouts were already optimized for right-to-left and adding the `android:supportsRtl="true"` attribute was the only thing needed to render them correctly.  The only thing I found that needed updating was the padding between the tag chips and ***Add tag...*** field at the bottom of the note editor.  The `ChipGroup` view was not honoring the `android:paddingEnd` attribute as it should.  As a workaround, I added a `View` with the same padding for its width between the two views.  See the screenshots below for illustration.

![add_rtl_support](https://user-images.githubusercontent.com/3827611/75200505-18d05600-5723-11ea-9098-d244cda769e5.png)

### Test
In order to see the layouts in right-to-left, a right-to-left language (e.g. Hebrew) must be selected at the system level _or_ the ***Force RTL layout direction*** setting can be enabled in ***Developer options***.  I've seen issues with the ***Force RTL layout direction*** setting giving false positives so I recommend selected a right-to-left language.

Note that even though a right-to-left language is selected, the app layout can show incorrectly _if left-to-right text is used_.  As an example, the ***Edit tags*** screen will show text to the left even when Hebrew is selected at the system level if the tags used English text.  Using Hebrew text will show the expected layout in that case.

Even though the test steps below only address the note editor since that's the only layout that was changed, it's best to smoke test the rest of the app to ensure no other issues are found.
1. Tap any note in list with no tags.
2. If note content does not overflow screen, notice ***Add tag...*** field is shown.
3. Notice no padding between ***Add tag...*** field and edge of note content.
4. If note content does overflow screen, scroll to view ***Add tag...*** field.
5. Notice no padding between ***Add tag...*** field and edge of note content.
6. Add tag to ***Add tag...*** field.
7. Notice no padding between tag and edge of note content.
8. Notice padding between tag chip and ***Add tag...*** field.

### Review
Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
`RELEASE-NOTES.txt` was updated in 5778e6d with:
> Added support for right-to-left languages and layouts
